src/Cake.Core/CakeException.cs

```diff
@@ -43,6 +43,15 @@ public CakeException(string message, Exception innerException)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CakeException"/> class.
+        /// </summary>
+        /// <param name="exitCode">Exit code</param>
+        public CakeException(int exitCode = 1)
+        {
+            System.Environment.Exit(exitCode);
```

Comment:
We don't want to exit from the constructor, as this will kill the build regardless of what error handling exists in the script.

What we want is additional constructors overloads with a exitCode parameter, having it stored in a read-only `int ExitCode` property where its default value is `1`.
Something like `public int ExitCode { get; } = 1;`

And constructors like
```` csharp
        public CakeException(int exitCode)
        {
            ExitCode = exitCode;
        }

        public CakeException(int exitCode, string message)
            : base(message)
        {
            ExitCode = exitCode;
        }

        public CakeException(int exitCode, string message, Exception innerException)
            : base(message, innerException)
        {
            ExitCode = exitCode;
        }
````
We want new overloads and not default parameters as we don't want to break ABI with existing code.

Reply:
I just updated the pr based on your comments. Sorry it took so long and sorry for the confusion